mlos_bench/mlos_bench/storage/util.py

```diff
@@ -32,7 +32,7 @@ def kv_df_to_dict(dataframe: pandas.DataFrame) -> Dict[str, Optional[TunableValu
     data = {}
     for _, row in dataframe.astype('O').iterrows():
         assert isinstance(row['parameter'], str)
-        assert row['value'] is None or isinstance(row['value'], (str, int, float))
+        assert isinstance(row['value'], TunableValueTypeTuple)
```

Comment:
Now when I look at it, I think we should throw an exception instead of using an `assert`. We should use the assertions for sanity checks of our own code; in this case, we check the validity of the external data (that can be produced by the user's script, i.e., not by our code)
```suggestion
        if not isinstance(row['value'], TunableValueTypeTuple):
            raise TypeError(f"Invalid column type: value:{type[row'value']}")
```